src/chatd.cpp

```diff
@@ -4454,7 +4596,9 @@ Idx Chat::msgIncoming(bool isNew, Message* message, bool isLocal)
             if (mHasMoreHistoryInDb)
             { //we have db history that is not loaded, so we determine the index
               //by the db, and don't add the message to RAM
-                idx = mDbInterface->getOldestIdx()-1;
+                chatd::Idx oldestIdx = mDbInterface->getOldestIdx();
+                assert(oldestIdx != CHATD_IDX_INVALID);
+                idx = oldestIdx - 1;
```

Comment:
```suggestion
                assert(mDbInterface->getOldestIdx() != CHATD_IDX_INVALID);
                idx = mDbInterface->getOldestIdx() - 1;
```

Reply:
In case assert doesn't fail, query will be executed twice to retrieve the same value

Reply:
The suggestion aimed to avoid the unnecessary allocation and copy into `oldestIdx`, but keeping the assert. It's true that it represents an extra DB query, but only in debug builds.
If you prefer to keep the allocation+assignment, I'm fine with it too :)